### PR TITLE
UCT/IB/DEVX: Check UAR support

### PR DESCRIPTION
**Pull request by Artemy-Mellanox**
## What
Fixing internal issue [3236372](https://redmine.mellanox.com/issues/3236372)

### CODE REVIEW

**Comment by yosefe** [comment]
Need to port it to v1.14.x as well

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@Artemy-Mellanox can u pls squash?

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 1]
```diff
@@ -941,6 +941,20 @@ static void uct_ib_mlx5_md_port_counter_set_id_init(uct_ib_mlx5_md_t *md)
     }
 }
 
+static int uct_ib_mlx5_check_uar(uct_ib_mlx5_md_t *md)
```
[the remark this answers is not in the document]
Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:944]: oops, missed

File: src/uct/ib/mlx5/dv/ib_mlx5dv_md.c [hunk 2]
```diff
@@ -1229,11 +1242,19 @@ static ucs_status_t uct_ib_mlx5_devx_md_open(struct ibv_device *ibv_device,
...
 err_lru_cleanup:
     uct_ib_mlx5_devx_mr_lru_cleanup(md);
+err_free_md:
     uct_ib_md_free(&md->super);
 err_free_context:
     uct_ib_md_device_context_close(ctx);
 err:
-    ucs_debug("%s: DEVX is not supported", ibv_get_device_name(ibv_device));
+    if ((status == UCS_ERR_UNSUPPORTED) && (md_config->devx == UCS_YES)) {
+        ucs_error("DEVX requested but not supported by %s",
```
[the remark this answers is not in the document]
Comment (Artemy-Mellanox) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:1251]: IMHO `enable_gpudirect_rdma` case should return `UCS_ERR_NO_DEVICE`, others are the absence of DEVX support for a reason whatever.

[the remark this answers is not in the document]
Comment (ivankochin) [reply src/uct/ib/mlx5/dv/ib_mlx5dv_md.c:1251]: So can we change `enable_gpudirect_rdma` case return code then?

**Pull request merged**